go/registry/api/api.go

```diff
@@ -1056,6 +1063,12 @@ func (g *Genesis) SanityCheck() error { // nolint: gocyclo
 		if !n.Consensus.ID.IsValid() {
 			return fmt.Errorf("registry: sanity check failed: node ID %s has an invalid consensus ID", n.ID.String())
 		}
+
+		if !n.P2P.ID.IsValid() {
+			return fmt.Errorf("registry: sanity check failed: node ID %s has an invalid P2P ID", n.ID.String())
+		}
+
+		// XXX: Validate addresses as well?
```

Comment:
Yes?

Reply:
Did you see the comment above about this breaking Genesis SanityCheck on a dump in existing deployments where validators have an invalid P2P ID / addresses? Should we allow invalid P2P keys/addresses for the next few versions in Genesis?

Reply:
Oh I see, ok in this case file an issue so that we don't forget when this can be fixed (probably on next redeploy after all nodes have either expired or updated their registrations?). Also reference the issue in the comment and say why there is currently no such check.

Reply:
done: https://github.com/oasislabs/oasis-core/issues/2428